roles/ceph-config/templates/ceph.conf.j2

```diff
@@ -151,6 +151,17 @@ filestore xattr use omap = true
 {# else, default is false #}
 {% endif %}
 {% endif %}
+{% if osd_objectstore == 'bluestore' %}
+[osd]
+osd memory target = {{ osd_memory_target }}
+{% if is_hci is defined %} # is_hci should be defined in group_vars/all.yml
+{% if num_osds > 0 and ansible_memtotal_mb * hci_safety_factor / num_osds  > osd_memory_target %} # hci_safety_factor is the safety factor for HCI deployments
+osd memory target = {{ ansible_memtotal_mb * hci_safety_factor / num_osds }}
+{% endif %}
```

Comment:
@neha-ojha something is wrong here, I think you want a `{% elif %}`?

Reply:
don't pay attention to my last comment, I think you are just missing a `{% endif %}`

Reply:
yeah, I noticed it when I wrote a pythonic version of these ifs :) fixing it

Reply:
@neha-ojha perhaps I'm missing something but as it is

If osd_objectstore = 'bluestore' and is_hci is defined, it means you are going to add `osd memory target` twice in the config, L156 + either L159 or L163

Reply:
I need the default value of `osd memory target` to be set to 4000000000. If those conditions are met, we will override this value else not.